bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttBrokerConnection.java

```diff
@@ -133,24 +131,28 @@ public void messageArrived(String topic, MqttMessage message) {
      * Create a new connection with the given name.
      *
      * @param name for the connection.
-     * @param url url string for the MQTT broker. Valid URL's are in the format: tcp://localhost:1883 or
-     *            ssl://localhost:8883
-     * @throws ConfigurationException
-     */
-    public MqttBrokerConnection(@NonNull String name, @NonNull String url, boolean textualConfiguredBroker)
-            throws ConfigurationException {
-        this.textualConfiguredBroker = textualConfiguredBroker;
-
-        if (name.isEmpty()) {
-            throw new ConfigurationException("No name for the broker set!");
+     * @param host A host name or address
+     * @param port A port or null to select the default port for a secure or insecure connection
+     * @param secure A secure connection
+     * @param clientId Set client id to use when connecting to the broker.
+     *            If none is specified, a default is generated. The client id cannot
+     *            be longer than 23 characters.
+     * @throws IllegalArgumentException
+     */
+    public MqttBrokerConnection(String host, @Nullable Integer port, boolean secure, @Nullable String clientId) {
+        this.host = host;
+        this.secure = secure;
+        if (clientId == null) {
+            clientId = MqttClient.generateClientId();
+            assert clientId != null;
```

Comment:
please remove or refactor

Reply:
Some asserts can not be removed with my eclipse, which was setup by oomph and the official installer. I get errors like "Null type mismatch (type annotations): required '@NonNull String' but this expression has type '@Nullable String'. The problem severity defaults to "error" for those null checks.